api/src/main/java/com/cloud/vm/schedule/VMSchedule.java

```diff
@@ -0,0 +1,49 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+package com.cloud.vm.schedule;
+
+import org.apache.cloudstack.api.Identity;
+import org.apache.cloudstack.api.InternalIdentity;
+
+public interface VMSchedule extends Identity, InternalIdentity {
+    public enum State {
+        Disabled, Enabled
+    }
+
+    public enum Action {
+        start, stop, forcestop, reboot
+    }
+
+    public String getUuid();
```

Comment:
Is this necesasry if VMSchedule extends Identity and InternalIdentity?

Reply:
yes

Reply:
Don't think `getUuid` needs to be defined if you are extending `Identity`